apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -0,0 +1,420 @@
+            Assert.NotEmpty(span.Labels);
+            Assert.Equal(span.Labels[Common.Labels.HttpMethod], "GET");
+            Assert.Equal(span.Labels[Common.Labels.HttpStatusCode], "200");
+        }
+
+        [Fact]
+        public async Task Trace_Label()
+        {
+            string testId = Utils.GetTestId();
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+
+            var builder = new WebHostBuilder().UseStartup<TraceTestNoBufferHighQpsApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                await client.GetAsync($"/Trace/TraceLabels/{testId}");
+            }
+
+            var spanName = TraceController.GetMessage(nameof(TraceController.TraceLabels), testId);
+            var trace = await GetTrace(spanName, startTime);
+
+            Assert.NotNull(trace);
+            Assert.Equal(2, trace.Spans.Count);
+            var span = trace.Spans.First(s => s.Name.StartsWith("Trace"));
+            Assert.Single(span.Labels);
+            Assert.Equal(span.Labels[TraceController.Label], TraceController.LabelValue);
+        }
+
+        [Fact]
+        public async Task Trace_StackTrace()
+        {
+            string testId = Utils.GetTestId();
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+
+            var builder = new WebHostBuilder().UseStartup<TraceTestNoBufferHighQpsApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                await client.GetAsync($"/Trace/TraceStackTrace/{testId}");
+            }
+
+            var spanName = TraceController.GetMessage(nameof(TraceController.TraceStackTrace), testId);
+            var trace = await GetTrace(spanName, startTime);
+
+            Assert.NotNull(trace);
+            Assert.Equal(2, trace.Spans.Count);
+            var span = trace.Spans.First(s => s.Name.StartsWith("Trace"));
+            Assert.Single(span.Labels);
+            Assert.Contains(nameof(TraceController), span.Labels[Common.Labels.StackTrace]);
+            Assert.Contains(nameof(TraceController.CreateStackTrace), span.Labels[Common.Labels.StackTrace]);   
+        }
+
+        [Fact]
+        public async Task Trace_Header()
+        {
+            string traceId = TraceIdFactory.Create().NextId();
+            ulong spanId = SpanIdFactory.Create().NextId();
+
+            string testId = Utils.GetTestId();           
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+            var builder = new WebHostBuilder().UseStartup<TraceTestNoBufferLowQpsApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                
+                var header = TraceHeaderContext.Create(traceId, spanId, true);
```

Comment:
Nit: as mentioned in a previous PR, I tend to dislike boolean literal arguments, so I would probably use `shouldTrace: true` here, but it's up to you.